js/latoken.js

```diff
@@ -0,0 +1,660 @@
+        let market = this.markets[symbol];
+        let key = 'quote';
+        let rate = market[takerOrMaker];
+        let cost = amount * rate;
+        let precision = market['precision']['price'];
+        if (side === 'sell') {
+            cost *= price;
+        } else {
+            key = 'base';
+            precision = market['precision']['amount'];
+        }
+        cost = this.decimalToPrecision (cost, ROUND, precision, this.precisionMode);
+        return {
+            'type': takerOrMaker,
+            'currency': market[key],
+            'rate': rate,
+            'cost': parseFloat (cost),
+        };
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        let response = await this.privateGetAccountBalances (params);
+        let result = {
+            'info': response,
+        };
+        for (let i = 0; i < response.length; i++) {
+            let balance = response[i];
+            let currency = balance['symbol'];
+            let account = {
+                'free': parseFloat (balance['available']),
+                'used': parseFloat (balance['frozen']),
+                'total': parseFloat (balance['amount']),
+            };
+            result[currency] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    parseBidAsk (bidask) {
+        let price = parseFloat (bidask['price']);
+        let amount = parseFloat (bidask['amount']);
+        return {
+            'price': price,
+            'amount': amount,
+        };
+    }
+
+    parseBidsAsks (bidasks) {
+        let res = [];
+        for (let i = 0; i < bidasks.length; i++) {
+            res.push (this.parseBidAsk (bidasks[i]));
+        }
+        return res;
+    }
+
+    parseOrderBook (orderbook, timestamp = undefined, asksKey = 'asks', bidsKey = 'bids') {
+        return {
+            'bids': (bidsKey in orderbook) ? this.parseBidsAsks (orderbook[bidsKey]) : null,
+            'asks': (asksKey in orderbook) ? this.parseBidsAsks (orderbook[asksKey]) : null,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'nonce': undefined,
+        };
+    }
+
+    async fetchOrderBook (symbol, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let request = {
+            'symbol': market['symbol'],
+        };
+        let timestamp = this.nonce ();
+        let response = await this.publicGetMarketDataOrderBook (this.extend (request, params));
+        let orderbook = this.parseOrderBook (response, timestamp);
+        orderbook['nonce'] = this.nonce ();
+        return orderbook;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        let symbol = this.findSymbol (this.safeString (ticker, 'symbol'), market);
+        let priceChangePercent = ((this.safeFloat (ticker, 'open')) / (this.safeFloat (ticker, 'open') + this.safeFloat (ticker, 'priceChange'))) * 100;
```

Comment:
We should not use `+` here, use `this.sum` instead, as described here: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

I suggest to break this line apart:

```
const open = this.safeFloat (ticker, 'open'); // reuse this variable below
const priceChange = this.safeFloat (ticker, 'priceChange');
let percentage = undefined;
if (open !== undefined) {
    if (priceChange !== undefined) {
        percentage = (open / this.sum (open, priceChange)) * 100;
    }
}
```

Reply:
thx!